low-level/framework/utils/sysfs_interface.py

```diff
@@ -138,7 +140,144 @@ def fetch_nw_cable_status(self, nw_interface_path, interface):
+        try:
+            host_dirpath = self.get_sys_dir_path('sas_host')
+            sas_hosts = os.listdir(host_dirpath)
+            sas_hosts = [host for host in sas_hosts
+                         if 'host' in host]
+        except Exception:
+            return []
+        return sas_hosts
+
+    def get_sas_port_list(self, host=""):
+        """
+            Return a list of SAS ports.
+
+            searches for ports from given host in /sys/class/sas_port
+            directory if directory exists else return an empty list.
+            eg: ['port-1:0', 'port-1:1', 'port-1:2', 'port-1:3']
+        """
+        sas_ports = []
+        try:
+            host_id = host.replace('host', '')
+            port_dirpath = self.get_sys_dir_path('sas_port')
+            sas_ports = os.listdir(port_dirpath)
+            sas_ports = [port for port in sas_ports
+                         if f'port-{host_id}' in port]
+        except Exception:
+            return []
+        return sas_ports
+
+    def get_sas_port_data(self, port_name):
+        """
+            Return SAS port data/info of given port.
+
+            eg: Given input port-1:0 returns
+                { "port_id":"sas_port-0",
+                  "state":"running",
+                  "sas_address":"0x500c0fff0a98b000"
+                }
+        """
+        port_data = {}
+        try:
+            host_id, port_id = port_name.replace('port-', '').split(':')
+            port_data_path = os.path.join(
+                             self.get_sys_dir_path('sas_end_device'),
+                             f'end_device-{host_id}:{port_id}',
+                             f'device/target{host_id}:0:{port_id}',
+                             f'{host_id}:0:{port_id}:0')
+            with open(os.path.join(port_data_path, 'state')) as cFile:
+                state = cFile.read().strip()
+
+            with open(os.path.join(port_data_path, 'sas_address')) as cFile:
+                sas_addr = cFile.read().strip()
+
+            port_data = {
+                    "port_id": f'sas_port-{port_id}',
+                    "state": state,
+                    "sas_address": sas_addr
+                }
+        except Exception:
+            return {}
+        return port_data
+
+    def get_phy_list_for_port(self, port):
+        """
+            Return list of SAS phys under given port.
+
+            eg: for given port-1:0 it might return
+                [ 'phy-1:0', 'phy-1:1', 'phy-1:2', 'phy-1:3']
+                returns an empty list if port-1:0 does not exist.
+        """
+        phys = []
+        try:
+            phy_list_dir = os.path.join(
+                            self.get_sys_dir_path('sas_port'),
+                            port, 'device')
+            phys = os.listdir(phy_list_dir)
+            phys = [phy for phy in phys if 'phy' in phy]
+            phys.sort(key=lambda phy: int(phy.split(':')[1]))
+        except Exception:
+            return []
+        return phys
```

Comment:
Exception may come with os, so you are making try-except. Thats fine.
But except `returns []`. At this time phys is also empty, right? Why don't we use `return phys` inside except block?

Reply:
Same, I see in other methods as well in this module. Please do the change.